src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/JsonCollectionConverter.cs

```diff
@@ -146,30 +147,42 @@ protected static JsonConverter<TElement> GetElementConverter(ref WriteStack stat
                 }
 
                 // Handle the metadata properties.
-                if (state.CanContainMetadata && state.Current.ObjectState < StackFrameObjectState.ReadMetadata)
+                if (state.Current.CanContainMetadata && state.Current.ObjectState < StackFrameObjectState.ReadMetadata)
                 {
-                    if (!JsonSerializer.TryReadMetadata(this, ref reader, ref state))
+                    if (!JsonSerializer.TryReadMetadata(this, jsonTypeInfo, ref reader, ref state))
```

Comment:
why can't TryReadMetadata read jsonTypeInfo from state?

Reply:
I suppose it could but most converters already store the reference as local. Thought I'd follow through on that convention, but no strong opinions other than that.